plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/launching/LocalAppEngineStandardLaunchShortcut.java

```diff
@@ -0,0 +1,105 @@
+
+package com.google.cloud.tools.eclipse.appengine.localserver.launching;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.ui.ILaunchShortcut2;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IServer;
+
+/**
+ * Provides support for run/debug on App Engine. Some of these methods are called frequently, such
+ * as to update tooltips.
+ */
+public class LocalAppEngineStandardLaunchShortcut implements ILaunchShortcut2 {
+
+  private LaunchHelper launcher = new LaunchHelper();
+
+  @Override
+  public void launch(ISelection selection, String launchMode) {
+    try {
+      IModule[] modules = launcher.asModules(selection);
+      launcher.launch(modules, launchMode);
+    } catch (CoreException ex) {
+      ErrorDialog.openError(null, "Unable to launch App Engine server", ex.getLocalizedMessage(),
+          ex.getStatus());
+    }
+  }
+
+  @Override
+  public void launch(IEditorPart editor, String launchMode) {
+    try {
+      IModule[] modules = launcher.asModules(editor);
+      launcher.launch(modules, launchMode);
+    } catch (CoreException ex) {
+      ErrorDialog.openError(null, "Unable to launch App Engine server", ex.getLocalizedMessage(),
+          ex.getStatus());
+    }
+  }
+
+  @Override
+  public ILaunchConfiguration[] getLaunchConfigurations(ISelection selection) {
+    IModule[] modules = null;
```

Comment:
please pull happy path into the try block so you don't have to split declaration and intiialization